src/store/modules/web3.ts

```diff
@@ -49,32 +48,34 @@ const actions = {
     auth = getInstance();
     commit('SET', { authLoading: true });
     await auth.login(connector);
-    if (auth.provider) {
-      auth.web3 = new Web3Provider(auth.provider);
+    if (auth.provider.value) {
+      auth.web3 = new Web3Provider(auth.provider.value);
       await dispatch('loadProvider');
     }
     commit('SET', { authLoading: false });
   },
   logout: async ({ commit }) => {
-    Vue.prototype.$auth.logout();
+    const auth = getInstance();
+    auth.logout();
     commit('WEB3_SET', { account: null, profile: null });
   },
   loadProvider: async ({ commit, dispatch }) => {
+    const auth = getInstance();
     try {
-      if (auth.provider.removeAllListeners) auth.provider.removeAllListeners();
-      if (auth.provider.on) {
-        auth.provider.on('chainChanged', async chainId => {
+      if (auth.web3.removeAllListeners) auth.web3.removeAllListeners();
```

Comment:
Can you just revert the changes here? This is part of another PR, need to do more test for it.

Reply:
Ok reverted it.